# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/tests/mochitest/test_getUserMedia_GC_MediaStream.html
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/tests/mochitest/test_getUserMedia_GC_MediaStream.html
--- a/dom/media/tests/mochitest/test_getUserMedia_GC_MediaStream.html	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/tests/mochitest/test_getUserMedia_GC_MediaStream.html	Wed Jul 31 07:58:17 2019 +0000
@@ -20,12 +20,12 @@
 
     let copies = new Array(numCopies).fill(0).map(() => copy(stream));
     ok(await SpecialStream.countUnderlyingStreams() > startStreams,
-        "MediaStream constructor creates more underlying streams");
+        "MediaStreamTrack constructor creates more underlying streams");
 
     copies = [];
     await new Promise(r => SpecialPowers.exactGC(r));
     is(await SpecialStream.countUnderlyingStreams(), startStreams,
-       "MediaStreams should have been collected");
+       "MediaStreamTracks should have been collected");
   }
 
   runTest(async () => {
@@ -34,19 +34,6 @@
     DISABLE_LOOPBACK_TONE = true;
 
     let gUMStream = await getUserMedia({video: true});
-    info("Testing GC of copy constructor");
-    await testGC(gUMStream, 10, s => new MediaStream(s));
-
-    info("Testing GC of track-array constructor");
-    await testGC(gUMStream, 10, s => new MediaStream(s.getTracks()));
-
-    info("Testing GC of empty constructor plus addTrack");
-    await testGC(gUMStream, 10, s => {
-      let s2 = new MediaStream();
-      s.getTracks().forEach(t => s2.addTrack(t));
-      return s2;
-    });
-
     info("Testing GC of track-array constructor with cloned tracks");
     await testGC(gUMStream, 10, s => new MediaStream(s.getTracks().map(t => t.clone())));
 